lib/pbench/agent/tool_group.py

```diff
@@ -75,7 +76,7 @@ def __init__(self, name):
 
         Raises BadToolGroup via the verify_tool_group() method on error.
         """
-        self.tg_dir = self.verify_tool_group(name)
+        self.tg_dir = ToolGroup.verify_tool_group(name, pbench_run)
```

Comment:
I'll make the change.

Moving this change to another PR (forthcoming).

Reply:
Well, it doesn't really work that way.  By invoking the static method with `self.verify_tool_group` an extra "self" argument is passed to the static method.  Only when I invoke it via the class name does it work correctly.

Reply:
I think you're likely conflating two separate issues.

In the CUT, there should be no difference between `ToolGroup.verify_tool_group()` and `self.verify_tool_group()`; however, when you're setting up a mock for the function in a unit test, you need to be very careful of how you define and reference the target (mock) function.

Since the function in the CUT is a `@staticmethod`, you need to make sure that the mock function either is a plain function or also is marked as a `@staticmethod`.  You can also make it explicit using the `staticmethod()` constructor.  If the mock is a regular member function, then it will end up being called without the `self` argument if you reference it with the class when you set the mock; if you reference it using an instance of a class when you set up the mock, then it will receive that class (not the class that it is invoked on!) as `self` when it is called.  (Python's binding rules never cease to surprise me.)

Reply:
Okay, the only way to get this to work is to use `staticmethod()` around the mock function.